src/Uno.UI/UI/Xaml/Controls/WebView/UnoWKWebView.iOSmacOS.cs

```diff
@@ -649,48 +641,73 @@ public override void DidClose(WKWebView webView)
 
 		private class WebViewNavigationDelegate : WKNavigationDelegate
 		{
-			private readonly Action<WKWebView, WKNavigation> _onNavigationCommited;
-			private readonly Action<WKWebView, WKNavigation> _onNavigationFinished;
-			private readonly Action<WKWebView, WKNavigation, NSError> _onNavigationFailed;
-			private readonly Action<WKWebView, WKNavigation> _onProvisionalNavigationStarted;
-			private readonly Action<WKWebView, WKNavigation, NSError> _onProvisionalNavigationFailed;
-			private readonly Func<Uri, bool> _onUnsupportedUriSchemeIdentified;
-
-			public WebViewNavigationDelegate(
-				Action<WKWebView, WKNavigation> onNavigationCommited,
-				Action<WKWebView, WKNavigation> onNavigationFinished,
-				Action<WKWebView, WKNavigation, NSError> onNavigationFailed,
-				Action<WKWebView, WKNavigation> onProvisionalNavigationStarted,
-				Action<WKWebView, WKNavigation, NSError> onProvisionalNavigationFailed,
-				Func<Uri, bool> onUnsupportedUriSchemeIdentified
-			)
+			/// <summary>
+			/// The reference to the parent UnoWKWebView class on which we invoke callbacks.
+			/// </summary>
+			private readonly UnoWKWebView _unoWKWebView;
```

Comment:
Not sure if necessary, but we could use a `WeakReference<UnoWKWebView>` here instead.

Reply:
Yeah I wasn't sure about that.
I changed the multiple Action<> to a single reference to the parent class to simplify the code. Since we were passing references to the instance methods, isn't it the same for the GC?

Reply:
Holding the references to object's delegate is not the same as holding ref to the object itseft.
I _think_ the latter prevents the object from being disposed properly, as we now have a circular reference (UnoWKWebView->NavigationDelegate->_unoWKWebView), but I am not sure on that.
At least, I've seen `WeakReference` quite often in uno when we pass reference around.

Reply:
A weak reference is indeed required here. You can add the `[Weak]` attribute on the field, it'll do the same.